generation/day_ahead.py

```diff
@@ -114,6 +115,13 @@ def import_day_ahead_generation(
     abort_if_data_empty(scheduled_generation)
     log.debug("Overall aggregated generation: \n%s" % scheduled_generation)
 
+    scheduled_generation = resample_if_needed(
+        scheduled_generation,
+        sensors["Scheduled generation"],
+        from_time,
```

Comment:
Isn't it more elegant and also safer if `resample_if_needed` gets start and from time from the actual series? As far as we know, these are what we requested, but maybe not what we are getting. Also, I believe we always want to resample the whole series if needed, not just a part.

Reply:
> Isn't it more elegant and also safer if resample_if_needed gets start and from time from the actual series? As far as we know, these are what we requested, but maybe not what we are getting.

Good point. This should be possible.

> Also, I believe we always want to resample the whole series if needed, not just a part.

Not sure what you mean.

Reply:
> Not sure what you mean.

Just another argument for not passing the dates, and getting them from the series

Reply:
> Just another argument for not passing the dates, and getting them from the series

Then I hope [my fix](https://github.com/SeitaBV/flexmeasures-entsoe/pull/21/commits/8844a3db98333d7ea3ce727545b5e9d99879fbbe) addresses your issue. Note that `(from_time, to_time)` runs from midnight to midnight, while `series.index` runs from midnight to either 23:00 (data before 2022-01-01) or 23:45 (data from 2022-01-01 onward). Therefore, the new code needs to add the event resolution of the received data on the right side of the time window. I'm inferring that event resolution from the inferred frequency of `series.index`.